src/utils/element.js

```diff
@@ -0,0 +1,122 @@
+import { isObject } from "./objects";
+
+const mappedPropertyFunctions = {
+  attribute: "setAttribute",
+  event: "addEventListener",
+};
+
+const initializeDomElement = (domElement) => {
+  const defaultDomElement = {
+    type: null,
+    classes: [],
+    attributes: {},
+    events: {},
+    html: "",
+    children: [],
+  };
+
+  return {
+    ...defaultDomElement,
+    ...domElement,
+  };
+};
+
+/**
+ *
+ * @param {*} element
+ * @param {*} children
+ * @param {*} callback
+ * @returns
+ */
+const appendChildren = (element, children) => {
+  if (!children.length) {
+    return element;
+  }
+
+  children.forEach((child) => {
+    const childElement = setupElement(child);
+    element.appendChild(childElement);
+  });
+
+  return element;
+};
+
+/**
+ *
+ * @param {*} element
+ * @param {*} classes
+ * @returns
+ */
+const setClassList = (element, classes) => {
+  if (!classes || !classes.length) {
+    return element;
+  }
+
+  classes.forEach((className) => element.classList.add(className));
+
+  return element;
+};
+
+/**
+ *
+ * @param {*} element
+ * @param {*} obj
+ * @param {*} property
+ * @returns
+ */
+const setElementKeys = (element, obj, property) => {
+  if (!isObject(obj)) {
+    return element;
+  }
+
+  const propertyFunction = mappedPropertyFunctions[property];
+
+  Object.entries(obj).forEach(([key, value]) =>
+    element[propertyFunction](key, value)
+  );
+
+  return element;
+};
+
+/**
+ *
+ * @param {*} element
+ * @param {*} html
+ * @returns
+ */
+const setHtml = (element, html = "") => {
+  if (html === "") {
+    return element;
+  }
+
+  const textNode = document.createTextNode(html);
+  element.appendChild(textNode);
+
+  return element;
+};
+
+/**
+ *
+ * @param {*} domElement
+ * @returns
+ */
+const setupElement = (domElement) => {
+  const initializedDomElement = initializeDomElement(domElement);
```

Comment:
I really like how this is set up. I've been reading Clean Code that Nexient sent us, and this is super clean. I like how everything is split out into smaller helper functions. It also makes me realize that I need to go and document all of my stuff because that was helpful when navigating.